### PR TITLE
Base server

### PR DESCRIPTION
**Pull request by FenrirUnbound**
This migrates the most basic API functionality over to a standalone module. It only explicit route it is equipped with is the build-in status module.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,13 @@
   "description": "API module for the Screwdriver CD service",
   "main": "index.js",
   "scripts": {
-    "pretest": "jscs .",
-    "test": "jenkins-mocha --recursive"
+    "test": "jenkins-mocha --recursive",
+    "start": "PORT=8666 node index.js",
+    "functional": "cucumber-js --format=pretty --format=json:${TEST_DIR}/cucumber_output.json; cat ${TEST_DIR}/cucumber_output.json | cucumber-junit > ${TEST_DIR}/func_test_results.xml"
+  },
+  "jshintConfig": {
```
Comment (petey) [inline package.json]: this isn't required. this was used internally.

File: package.json [hunk 2]
```diff
@@ -4,8 +4,13 @@
   "description": "API module for the Screwdriver CD service",
   "main": "index.js",
   "scripts": {
-    "pretest": "jscs .",
```
Comment (petey) [inline package.json]: You aren't linting at all anymore.

File: package.json [hunk 3]
```diff
@@ -20,18 +25,32 @@
   "license": "BSD-3-Clause",
   "author": "Darren Matsumoto <aeneascorrupt@gmail.com>",
   "contributors": [
-    "Dao Lam <d2lam@yahoo-inc.com>",
+    "Dao Lam <daolam112@gmail.com>",
```
Comment (stjohnjohnson) [inline package.json]: Please update the generator as well

File: plugins/logging.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    register: require('good'),
+    options: {
+        ops: {
+            interval: 1000
+        },
+        reporters: {
+            console: [{
+                module: 'good-squeeze',
+                name: 'Squeeze',
+                args: [{ log: '*', response: '*' }]
```
Comment (stjohnjohnson) [inline plugins/logging.js]: Please output `error: '*'`

File: plugins/swagger.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    register: require('hapi-swagger'),
+    options: {
+        info: {
+            title: 'hapi-engine documentation',
```
Comment (stjohnjohnson) [inline plugins/swagger.js]: Screwdriver API Documentation

File: test/lib/server.test.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        });
+
+        it('does it with a different port', (done) => {
+            processEnvMock.PORT = 12347;
+
+            hapiEngine((error, server) => {
+                Assert.notOk(error);
+                server.inject({
+                    method: 'GET',
+                    url: '/v3/status'
```
Comment (stjohnjohnson) [inline test/lib/server.test.js]: Should status be at /v3 or at /status?

File: package.json (hunk 1 above)
Comment (FenrirUnbound) [inline package.json]: this we can remove

File: package.json (hunk 2 above)
Comment (FenrirUnbound) [inline package.json]: The linting upgrade is a big change on its own, so we'll make a separate PR for it

File: package.json (hunk 3 above)
Comment (FenrirUnbound) [inline package.json]: Updated with screwdriver-cd/generator-screwdriver#7

File: test/lib/server.test.js (hunk 6 above)
Comment (FenrirUnbound) [inline test/lib/server.test.js]: The initial requirement was with a `/v3` prefix

Comment (stjohnjohnson) [inline test/lib/server.test.js]: Got it.

File: package.json (hunk 2 above)
Comment (FenrirUnbound) [inline package.json]: j/k we're doing it all at once 😺

**Comment by stjohnjohnson** [comment]
👍  
yup

File: features/step_definitions/server.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+module.exports = () => {
+    this.Given(/^a running hapi server$/, (callback) => {
+        // Write code here that turns the phrase above into concrete actions
+        this.instance = process.env.INSTANCE;
+        callback(null);
+    });
+
+    this.When(/^I access a status endpoint$/, (callback) => {
+        // Write code here that turns the phrase above into concrete actions
```
Comment (jer) [inline features/step_definitions/server.js]: These comments seem unnecessary

**Comment by d2lam** [comment]
👍
